features/mbedtls/targets/TARGET_STM/aes_alt.c

```diff
@@ -0,0 +1,495 @@
+/*
+ *  Hardware aes collector for the STM32F4 family
+ *******************************************************************************
+ * Copyright (c) 2017, STMicroelectronics
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+#if defined(MBEDTLS_AES_ALT)
+
+#include <stdio.h>
+#include "cmsis.h"
+#include "string.h"
+#include "aes.h"
+CRYP_HandleTypeDef hcryp_aes;
```

Comment:
you are totally right about the thread safety. I already have an issue with the CTR-DRBG self-test that is calling 2 instances of aes. I will rework it and declare CRY_HandleTypedef as part of the mbedtls_context_aes and remove the unnecessary fields (nr ...).
No more global (or even static )